### PR TITLE
Improve the document for cEOS image downloading

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
The cEOS image downloading has been improved in PR #13840. This change updated the document accordingly.

#### How did you do it?
Improved the cEOS downloading documentation.

### CODE REVIEW

File: docs/testbed/README.testbed.VsSetup.md [hunk 1]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
...
+   2. Place the image file under subfolder `images` of the location determined by `root_path` variable in `ansible/group_vars/vm_host/main.yml`:
+
+      Assume `root_path` is `veos-vm`:
+
+      ```bash
+      cp cEOS64-lab-4.29.3M.tar ~/veos-vm/images/
+      ```
+      The Ansible playbook for deploying testbed topology will automatically use the manually prepared image file from this location.
+
+   #### Option 1.2: Host the cEOS image file on a HTTP server
```
Comment (opcoder0) [inline docs/testbed/README.testbed.VsSetup.md:89]: This section is under **2**. I suppose it should be 2.2?

File: docs/testbed/README.testbed.VsSetup.md [hunk 2]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
...
+      ```
+      Or:
+      ```yaml
+      ceos_image_url:
+         - http://192.168.1.10/cEOS64-lab-4.29.3M.tar
+      skip_ceos_image_downloading: false
+      ```
 
 ### Option 3: Use SONiC image as neighboring devices
 You need to prepare a sound SONiC image `sonic-vs.img` in `~/veos-vm/images/`. We don't support to download sound sonic image right now, but for testing, you can also follow the section [Download the sonic-vs image](##download-the-sonic-vs-image) to download an available image and put it into the directory `~/veos-vm/images`
```
Comment (opcoder0) [inline docs/testbed/README.testbed.VsSetup.md:119]: Nit rephrase:

You need to create a valid SONiC image named `sonic-vs.img` in the `~/veos-vm/images/` directory. Currently, we don’t support downloading a pre-built SONiC image. However, for testing purposes, you can refer to the section [Download the sonic-vs image](##download-the-sonic-vs-image) to obtain an available image and place it in the `~/veos-vm/images/` directory.

File: docs/testbed/README.testbed.WANSetup.md [hunk 3]
```diff
@@ -30,33 +30,64 @@ First, we need to prepare the host where we will be configuring the virtual test
...
-```
-**Note**: *For time being, the image might be updated, in that case you can't download the same version of image as in the instruction,
-please download the corresponding version(following [Arista recommended release](https://www.arista.com/en/support/software-download#datatab300)) of image and import it to your local docker repository.
-The actual image version that is needed in the installation process is defined in the file [ansible/group_vars/all/ceos.yml](../../ansible/group_vars/all/ceos.yml), make sure you modify locally to keep it up with the image version you imported.*
+### Option 1: cEOS (container-based) image (recommended)
 
-**Note**: *Please also notice the type of the bit for the image, in the example above, it is a standard 32-bit image. Please import the right image as your needs.*
+1. **Prepare folder for image files on test server**
+
+    Create subfolder `images` under `root_path` defined in the `ansible/group_vars/vm_host/main.yml` file. Assume `root_path` is `veos-vm`:
```
Comment (opcoder0) [inline docs/testbed/README.testbed.WANSetup.md:37]: Nit re-phrase:

Create a subfolder called `images` inside the `root_path` directory defined in `ansible/group_vars/vm_host/main.yml` file. For instance, if `root_path` is set to `veos-vm`, you should run the following command:

File: docs/testbed/README.testbed.WANSetup.md [hunk 4]
```diff
@@ -30,33 +30,64 @@ First, we need to prepare the host where we will be configuring the virtual test
...
+    ```bash
+    mkdir -p ~/veos-vm/images
+    ```
+
+2. **Prepare the cEOS image file**
+
+   #### Option 1.1: Manually download cEOS image
+
+   1. Obtain the cEOS image from [Arista's software download page](https://www.arista.com/en/support/software-download).
+   2. Place the image file under subfolder `images` of the location determined by `root_path` variable in `ansible/group_vars/vm_host/main.yml`:
```
Comment (opcoder0) [inline docs/testbed/README.testbed.WANSetup.md:48]: Place the image file in the `images` subfolder located within the directory specified by the `root_path` variable in the `ansible/group_vars/vm_host/main.yml` file.

File: docs/testbed/README.testbed.WANSetup.md [hunk 5]
```diff
@@ -30,33 +30,64 @@ First, we need to prepare the host where we will be configuring the virtual test
...
+
+      Assume `root_path` is `veos-vm`:
```
Comment (opcoder0) [inline docs/testbed/README.testbed.WANSetup.md:50]: Nit rephrase:

Assuming you set `root_path` to `veos-vm`, you should run the following command:

File: docs/testbed/README.testbed.WANSetup.md [hunk 6]
```diff
@@ -30,33 +30,64 @@ First, we need to prepare the host where we will be configuring the virtual test
...
+      skip_ceos_image_downloading: false
+      ```
+      Or:
+      ```yaml
+      ceos_image_url:
+         - http://192.168.1.10/cEOS64-lab-4.29.3M.tar
+      skip_ceos_image_downloading: false
+      ```
+
+**Note**: *For time being, the image might be updated, in that case you can't download the same version of image as in the instruction,
```
Comment (opcoder0) [inline docs/testbed/README.testbed.WANSetup.md:86]: Nit rephrase:

**Note** When downloading, the version specified above might be outdated or unavailable. Please check the [Arista recommended release page](https://www.arista.com/en/support/software-download#datatab300) to obtain the latest recommended image and import it into your local Docker registry

File: docs/testbed/README.testbed.WANSetup.md [hunk 7]
```diff
@@ -30,33 +30,64 @@ First, we need to prepare the host where we will be configuring the virtual test
...
+please download the corresponding version(following [Arista recommended release](https://www.arista.com/en/support/software-download#datatab300)) of image and import it to your local docker repository.
+The actual image version that is needed in the installation process is defined in the file [ansible/group_vars/all/ceos.yml](../../ansible/group_vars/all/ceos.yml), make sure you modify locally to keep it up with the image version you imported.*
+
+**Note**: *Please also notice the type of the bit for the image, in the example above, it is a standard 32-bit image. Please import the right image as your needs.*
```
Comment (opcoder0) [inline docs/testbed/README.testbed.WANSetup.md:90]: Nit rephrase: 

**Note:** Please be aware of the image's CPU architecture (32 vs 64-bit). In the example above, it is a standard 64-bit cEOS image. Ensure you import the correct image according to your requirements.

**Review by opcoder0 — APPROVED** [review]
Hi @wangxin thank you for the update. LGTM. All my suggestions are minor and involve rephrasing only.

File: docs/testbed/README.testbed.VsSetup.md [hunk 8]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
 ## Download an VM image
 We currently support EOS-based or SONiC VMs to simulate neighboring devices in the virtual testbed, much like we do for physical testbeds. To do so, we need to download the image to our testbed host.
 
+**Prepare folder for image files on testbed host**
+
+Location for storing image files on testbed host is defined by variable `root_path` in the `ansible/group_vars/vm_host/main.yml` file. Please update this variable to the location you planned on testbed host. It can be absolute path or relative path. If use relative path, it will be relative to the home folder of the user will be used for accessing testbed host.
```
[the remark this answers is not in the document]
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:43]: Updated

File: docs/testbed/README.testbed.VsSetup.md [hunk 9]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
...
-ceos_image_url: "http://example1.com/cEOS64-lab-4.29.3M.tar"
-```
-The `ceos_image_url` variable also can be a list of URLs, for example:
-```
-ceos_image_url:
-  - "http://example1.com/cEOS64-lab-4.29.3M.tar"
-  - "http://example2.com/cEOS64-lab-4.29.3M.tar"
-```
-The playbook will try to download the image from the URLs in the list one by one until it succeeds.
+    Create subfolder `images` under `root_path` defined in the `ansible/group_vars/vm_host/main.yml` file. Assume `root_path` is `veos-vm`:
```
[the remark this answers is not in the document]
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:69]: Updated

File: docs/testbed/README.testbed.VsSetup.md [hunk 10]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
...
+
+    ```bash
+    mkdir -p ~/veos-vm/images
+    ```
+
+2. **Prepare the cEOS image file**
+
+   #### Option 1.1: Manually download cEOS image
```
[the remark this answers is not in the document]
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:77]: Good catch!

File: docs/testbed/README.testbed.VsSetup.md [hunk 11]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
...
+
+   1. Obtain the cEOS image from [Arista's software download page](https://www.arista.com/en/support/software-download).
+   2. Place the image file under subfolder `images` of the location determined by `root_path` variable in `ansible/group_vars/vm_host/main.yml`:
```
[the remark this answers is not in the document]
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:80]: Updated

File: docs/testbed/README.testbed.VsSetup.md [hunk 12]
```diff
@@ -38,36 +38,82 @@ sudo -H ./setup-management-network.sh
...
+
+      Assume `root_path` is `veos-vm`:
```
[the remark this answers is not in the document]
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:82]: Updated

File: docs/testbed/README.testbed.VsSetup.md (hunk 1 above)
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:89]: Updated

File: docs/testbed/README.testbed.VsSetup.md (hunk 2 above)
Comment (wangxin) [reply docs/testbed/README.testbed.VsSetup.md:119]: updated

File: docs/testbed/README.testbed.WANSetup.md (hunk 3 above)
Comment (wangxin) [reply docs/testbed/README.testbed.WANSetup.md:37]: Updated/

**Comment by wangxin** [comment]
Thanks @opcoder0  for the suggestions. I have updated the document accordingly.

**Review by opcoder0 — APPROVED** [review]
LGTM!

**Pull request merged**